components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
-            cam_in%lhf(:,:) , cam_in%cflx )
-#else
-       call qneg4('TPHYSAC '       ,lchnk               ,ncol  ,ztodt ,               &
-            state%q(1,pver,1),state%rpdel(1,pver) ,cam_in%shf ,         &
-            cam_in%lhf , cam_in%cflx )
-#endif 
+  call pbuf_get_field(pbuf, tini_idx, tini)
+  call pbuf_get_field(pbuf, qini_idx, qini)
+  call pbuf_get_field(pbuf, cldliqini_idx, cldliqini)
+  call pbuf_get_field(pbuf, cldiceini_idx, cldiceini)
 
-    end if 
-#endif
+  call pbuf_get_field(pbuf, pbuf_get_index('CLD'),cld,start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
+  call pbuf_get_field(pbuf, pbuf_get_index('AST'),ast,start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
 
-    call check_qflx(state, tend, "PHYAC02", nstep, ztodt, cam_in%cflx(:,1))
+  !-----------------------------------------------------------------------------
+  ! emissions of aerosols and gas-phase chemistry constituents at surface
+  !-----------------------------------------------------------------------------
+  if (l_tracer_aero) then
+    call chem_emissions( state, cam_in )
+  end if
 
-!!== KZ_WCON
+  !-----------------------------------------------------------------------------
+  ! get nstep and zero array for energy checker
+  !-----------------------------------------------------------------------------
+  zero = 0._r8
+  nstep = get_nstep()
+  call check_tracers_init(state, tracerint)
 
-    call t_stopf('tphysac_init')
+  ! Check if LHF exceeds the total moisture content of the lowest layer
+  call qneg4('TPHYSAC ', lchnk, ncol, ztodt, &
+              state%q(1,pver,1), state%rpdel(1,pver), &
+              cam_in%shf, cam_in%lhf, cam_in%cflx )
 
-if (l_tracer_aero) then
-    !===================================================
-    ! Source/sink terms for advected tracers.
-    !===================================================
-    call t_startf('adv_tracer_src_snk')
-    ! Test tracers
+  !-----------------------------------------------------------------------------
+  ! Source/sink terms for advected tracers.
+  !-----------------------------------------------------------------------------
+  if (l_tracer_aero) then
 
     call tracers_timestep_tend(state, ptend, cam_in%cflx, cam_in%landfrac, ztodt)      
     call physics_update(state, ptend, ztodt, tend)
-    call check_tracers_chng(state, tracerint, "tracers_timestep_tend", nstep, ztodt,   &
-         cam_in%cflx)
+    call check_tracers_chng(state, tracerint, "tracers_timestep_tend", nstep, ztodt, cam_in%cflx)
 
     call aoa_tracers_timestep_tend(state, ptend, cam_in%cflx, cam_in%landfrac, ztodt)      
 
     call physics_update(state, ptend, ztodt, tend)
-    call check_tracers_chng(state, tracerint, "aoa_tracers_timestep_tend", nstep, ztodt,   &
-         cam_in%cflx)
+    call check_tracers_chng(state, tracerint, "aoa_tracers_timestep_tend", nstep, ztodt, cam_in%cflx)
+
+    ! add tendency from aircraft emissions
+    call co2_cycle_set_ptend(state, pbuf, ptend)
```

Comment:
This tendency is duplicated, so it looks like we are adding tendency from aircraft emissions TWICE!

Reply:
Good catch, my bad, will remove.